plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,243 @@
+    add: ['test.txt', 'txt.test']
+    empty_commit: fail
+
+- name: Empty commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing empty commit
+    empty_commit: allow
+    add: ['.']
+
+- name: Skip if nothing to commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: Skip my amazing empty commit
+    empty_commit: skyp
+    add: ['.']
+'''
+
+RETURN = '''
+output:
+    description: list of git cli command stdout
+    type: list
+    returned: always
+    sample: [
+        "[master 99830f4] Remove [ test.txt, tax.txt ]\n 4 files changed, 26 insertions(+)..."
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def git_add(module):
+
+    add = module.params.get('add')
+    path = module.params.get('path')
+
+    if not add:
+        add = ["."]
+    # "git add do_not_exist" -> rc: 128
+    # "git add i_exist"      -> rc: 0
+    # "git add ."            -> rc: 0
+    add_cmds = [
+        'git',
+        'add',
+    ]
+    for item in add:
+        add_cmds.insert(len(add_cmds), item)
+
+    rc, _output, error = module.run_command(add_cmds, cwd=path)
+
+    if rc == 128:
+        module.fail_json(msg=error)
+
+    if rc == 0:
+        # no output returned when rc = 0
+        return
+
+
+def git_commit(module):
+
+    def git_commit_run(cmd, path):
+
+        rc, output, error = module.run_command(cmd, cwd=path)
+
+        if rc == 1:
+            module.fail_json(msg=error)
+        if rc == 0:
+            return output
+
+    result = dict(changed=False)
+    empty_commit = module.params.get('empty_commit')
+    comment = module.params.get('comment')
+    path = module.params.get('path')
+    cmd_porcelain = ['git', 'commit', '--porcelain']
+
+    # rc is always 1 for "git commit --porcelain"
+    _rc, output, error = module.run_command(cmd_porcelain, cwd=path)
+
+    if error:
+        module.fail_json(msg=error)
+
+    # do not allow empty commit with empty_commit=fail
+    if empty_commit == 'fail' and not output:
+        module.fail_json(msg='No empty commit allowed with empty_commit=fail')
+
+    # add and commit if output and empty_commit=fail
+    if empty_commit == 'fail' and output:
+
+        git_add(module)
+
+        commit_cmds = [
+            'git',
+            'commit',
+            '-m',
+            '"{0}"'.format(comment),
+        ]
+
+        output = git_commit_run(commit_cmds, path)
+
+        if output:
+            result.update(
+                git_commit=output,
+                changed=True
+            )
+
+            return result
+
+    # if empty_commit=allow skip git_add()
+    if empty_commit == 'allow' and not output:
+        commit_cmds = [
+            'git',
+            'commit',
+            '--allow-empty',
+            '-m',
+            '"{0}"'.format(comment),
+        ]
```

Comment:
This looks like unnecessary code repeat. Why not have this code once, and add additional parameters when needed?